src/installer/corehost/cli/bundle/manifest.h

```diff
@@ -16,16 +17,28 @@ namespace bundle
     {
     public:
         manifest_t()
-            : m_need_extraction(false) {}
+            : m_files_need_extraction(false)
+            , m_netcoreapp3_compat_mode(false)
+        {
+        }
 
         std::vector<file_entry_t> files;
 
-        static manifest_t read(reader_t &reader, int32_t num_files);
+        static manifest_t read(reader_t &reader, header_t &header);
 
-        bool files_need_extraction() { return m_need_extraction; }
+        bool files_need_extraction() const
+        {
+            return m_files_need_extraction;
+        }
+
+        bool is_netcoreapp3_compat_mode() const
+        {
+            return m_netcoreapp3_compat_mode;
+        }
 
     private:
-        bool m_need_extraction;
+        bool m_files_need_extraction;
+        bool m_netcoreapp3_compat_mode;
```

Comment:
Actually, what exactly is netcoreapp3 compat mode / how far does 'compat' go? The comments for `header_flags_t` call out extraction and `AppContext.BaseDirectory`. We also modified some other behaviours (e.g. `GetCommandLineArgs`, `APP_CONTEXT_DEPS_FILES`) - those aren't part of compat mode?

Reply:
My understanding was that those behaviors should be based on whether the containing assembly/file was loaded from the bundle. If it works based on whether the file was merely present in the bundle, that could be a problem.

Reply:
The APIs on `Assembly` are based on if the assembly was loaded from the bundle, but `Environment.GetCommandLineArgs` and the `APP_CONTEXT_DEPS_FILES` property are not tied to an assembly. I believe they are just based on whether or not we are running a bundle. (Not sure if there have been other behaviour changes that fall into the same category.)

Reply:
Yeah I see what you mean. It's probably worth auditing every call to `is_single_file_bundle`, since that API alone doesn't differentiate between a bundle for extraction, and a bundle for in-memory load.